framework/vkdf-init.cpp

```diff
@@ -64,13 +70,21 @@ get_required_extensions(VkdfContext *ctx, bool enable_validation)
    }
 
    ctx->inst_extensions = g_new(char *, ctx->inst_extension_count);
-   for (uint32_t i = 0; i < platform_extension_count; i++)
-      ctx->inst_extensions[i] = g_strdup(platform_extensions[i]);
-
+   uint32_t i = 0;
    if (enable_validation) {
-      ctx->inst_extensions[ctx->inst_extension_count - 1] =
+      ctx->inst_extensions[i++] =
          g_strdup(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
    }
+
+   if (!ctx->no_swapchain) {
+      ctx->inst_extensions[i++] =
+         g_strdup(VK_KHR_SURFACE_EXTENSION_NAME);
+   }
+
+   for (uint32_t c = 0; c < platform_extension_count; c++) {
```

Comment:
let's assert that platform_extension_count + i < ctx->inst_extension_count before this loop

Reply:
added but with <=